compiler/parser.cpp

```diff
@@ -717,189 +721,182 @@ set_extension(char* filename, const char* extension, int force) {
+    sc_showincludes = opt_showincludes.value();
+    sc_listing = opt_listing.value();
+    sc_compression_level = opt_compression.value();
+    sc_tabsize = opt_tabsize.value();
+    sc_use_new_parser = opt_new_parser.value();
+    sc_needsemicolon = opt_semicolons.value();
+
+    if (opt_codeversion.hasValue()) {
+        switch (opt_codeversion.value()) {
+            case 13:
+            case 12:
+                pc_must_drop_stack = false;
+                /* Fallthrough */
+            case 10:
+                pc_code_version = opt_codeversion.value();
+                break;
+            default:
+                fprintf(stderr, "unknown code version: %d\n", opt_codeversion.value());
+                exit(1);
+        }
+    }
+
+    pc_optimize = opt_optlevel.value();
+    if (pc_optimize < sOPTIMIZE_NONE || pc_optimize >= sOPTIMIZE_NUMBER ||
+        pc_optimize == sOPTIMIZE_NOMACRO)
+    {
+        Usage(parser, argc, argv);
+    }
+
+    if (opt_prefixfile.hasValue())
+        strlcpy(pname, opt_prefixfile.value().chars(), _MAX_PATH);
+    if (opt_outputfile.hasValue())
+        strlcpy(oname, opt_outputfile.value().chars(), _MAX_PATH);
+
+    if (opt_verbosity.hasValue()) {
+        if (isdigit(*opt_verbosity.value().chars()))
+            verbosity = atoi(opt_verbosity.value().chars());
+        else
+            verbosity = 2;
+    }
+
+    sc_asmfile = opt_assembly.value();
+    if (sc_asmfile && verbosity > 1)
+        verbosity = 1;
+
+    if (opt_active_dir.hasValue()) {
+        const char* ptr = opt_active_dir.value().chars();
 #if defined dos_setdrive
-                    if (ptr[1] == ':')
-                        dos_setdrive(toupper(*ptr) - 'A' + 1); /* set active drive */
+        if (ptr[1] == ':')
+            dos_setdrive(toupper(*ptr) - 'A' + 1); /* set active drive */
 #endif
-                    if (chdir(ptr)) {
-                        fprintf(stderr, "chdir failed: %s\n", strerror(errno));
-                        exit(1);
-                    }
-                    break;
-                case 'e':
-                    strlcpy(ename, option_value(ptr, argv, argc, &arg),
-                            _MAX_PATH); /* set name of error file */
-                    break;
-                case 'E':
-                    sc_warnings_are_errors = true;
-                    break;
+            if (chdir(ptr)) {
+                fprintf(stderr, "chdir failed: %s\n", strerror(errno));
+                exit(1);
+            }
+    }
+
+    if (opt_error_file.hasValue())
+        strlcpy(ename, opt_error_file.value().chars(), _MAX_PATH);
+
 #if defined __WIN32__ || defined _WIN32 || defined _Windows
-                case 'H':
-                    hwndFinish = (HWND)atoi(option_value(ptr, argv, argc, &arg));
-                    if (!IsWindow(hwndFinish))
-                        hwndFinish = (HWND)0;
-                    break;
+    hwndFinish = (HWND)atoi(opt_hwnd.value().chars());

```

Comment:
This crashes on windows when not passing the `-H` option due to not having a default value.

Reply:
Thanks, good catch, will fix this.